dbms/src/Functions/currentUser.cpp

```diff
@@ -0,0 +1,54 @@
+#include <Functions/IFunction.h>
+#include <Functions/FunctionFactory.h>
+#include <Interpreters/Context.h>
+#include <DataTypes/DataTypeString.h>
+
+
+namespace DB
+{
+
+class FunctionCurrentUser : public IFunction
+{
+    const String user_name;
+
+public:
+    static constexpr auto name = "currentUser";
+    static FunctionPtr create(const Context & context)
+    {
+        return std::make_shared<FunctionCurrentUser>(context.getClientInfo().current_user);
+    }
+
+    explicit FunctionCurrentUser(const String & user_name_) : user_name{user_name_}
+    {
+    }
+
+    String getName() const override
+    {
+        return name;
+    }
+    size_t getNumberOfArguments() const override
+    {
+        return 0;
+    }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & /*arguments*/) const override
+    {
+        return std::make_shared<DataTypeString>();
+    }
+
+    bool isDeterministic() const override { return false; }
+
+    void executeImpl(Block & block, const ColumnNumbers &, size_t result, size_t input_rows_count) override
+    {
+        block.getByPosition(result).column = DataTypeString().createColumnConst(input_rows_count, user_name);
```

Comment:
One subtle detail... we need to materialize the column (convertToFullColumn), because in distributed setup, the function can be executed on every remote server and the cluster can be configured with different users for different servers. And if the column is const, it must have the same value in every block in data stream.

Reply:
BTW, what user should we get: current user or initial user? If we will get initial user, we can leave this column constant. And probably this is what is expected.

Reply:
In that case, it must be resembled in docs.

Reply:
I think this function should return initial user. But that case, name is a bit confusing. Checked distributed query behaviour:
```
SELECT 
    user, 
    initial_user
FROM remote('clickhouseuds3.mlan', system.processes) 
WHERE query LIKE '%sleep%'
LIMIT 1

Row 1:
──────
user:         default
initial_user: krash
```

Reply:
Do you approve using `initial_user` in this function?

Reply:
Yes. The name is Ok.